Include/internal/pycore_object.h

```diff
@@ -627,46 +625,26 @@ PyObject * _PyObject_GetInstanceAttribute(PyObject *obj, PyDictValues *values,
 #endif
 
 typedef union {
-    PyObject *dict;
-    /* Use a char* to generate a warning if directly assigning a PyDictValues */
-    char *values;
-} PyDictOrValues;
+    PyDictObject *dict;
```

Comment:
Maybe this should just be typed to `PyObject *`?  It seems like it causes more casts then it saves.

Reply:
It probably does, but most of those casts are from functions that should return `PyDictObject *` but return `PyObject *`. 
I'd like to push people towards using the correct types, even it is a bit inconvenient sometimes.

Reply:
https://github.com/python/cpython/pull/116115/commits/ccceffb4e1bcdbfbb0983b4e9d08f561d062bc15